.dockerignore

```diff
@@ -0,0 +1,32 @@
+# ignore docker config
+docker-compose.yml
+Dockerfile
+.dockerignore
+nginx.conf
+
+# ignore repo docs
+EDITING.md
+README.md
+LICENSE.txt
+.git*
+
+# ignore secrets / CI logic
+build_key.enc
+push-build.sh
+.travis.yml
+
+# ignore unnecessary certbot files
+_docs
+!_docs/tools
+!_docs/acme
+!_docs/certbot
+!_docs/certbot-*
+!_docs/letsencrypt-auto-source
```

Comment:
> I tested this PR and in addition to my fairly minor inline comments, I noticed that one of the problems I described at #621 is unresolved. The problem is when I try to access URLs like http://localhost:4000/lets-encrypt/snap-other, I get a 404. This may simply be due to ignoring `nginx.conf` in `.dockerignore`, but I didn't have a chance to test this hypothesis.

Weird, I'm not seeing this issue. Are you bringing the container up with docker-compose, or just docker? The compose file includes an [nginx service](https://github.com/certbot/website/blob/master/docker-compose.yml#L3-L11) which brings `nginx.conf` in as a bind mount, and this should be the container serving localhost:4000 requests. Is that what you see?

Reply:
I removed those `_docs` entries from .dockerignore and changed it to only ignore `_docs/venv*` now. I'd initially set the ignore config up for _docs because that directory was sitting at 1.4G, but looking closer it was almost all from local venv files that the container needs to install for itself anyway. Those lines starting with `!` in .dockerignore take paths *off* the ignore list btw, which is how tools/ and letsencrypt-auto-source/ etc. were built into the image despite the rest of the repo being excluded.